src/MEDS_DEV/models/cehrbert/generate_cehrbert_predictions.py

```diff
@@ -0,0 +1,42 @@
+import logging
+from pathlib import Path
+
+import hydra
+import polars as pl
+from meds_evaluation.schema import validate_binary_classification_schema
+from omegaconf import DictConfig
+
+logger = logging.getLogger(__name__)
+
+CONFIG = Path(__file__).parent / "_config.yaml"
+
+
+@hydra.main(version_base=None, config_path=str(CONFIG.parent.resolve()), config_name=CONFIG.stem)
+def main(cfg: DictConfig) -> None:
+    predictions_fp = Path(cfg.predictions_fp)
+    predictions_fp.parent.mkdir(parents=True, exist_ok=True)
+    # Get the output dir
+    model_finetuned_dir = Path(cfg.model_initialization_dir)
+    # Infer the task label
+    task_label_name = Path(cfg.labels_dir).name
+    # Fine-tuned model dir
+    test_prediction_output_dir = model_finetuned_dir / task_label_name / "test_predictions"
+    if not test_prediction_output_dir.exists():
+        raise RuntimeError(
+            "The previous fine-tuning step did not generate the predictions for the held-out set."
+        )
+
+    predictions = pl.read_parquet(list(test_prediction_output_dir.rglob("*.parquet")))
+    # TODO: calibrate the classifier and generate better boolean predictions
+    predictions = predictions.with_columns(
+        predictions["predicted_boolean_probability"].cast(pl.Float64).alias("predicted_boolean_probability"),
+        (predictions["predicted_boolean_probability"] > 0.5).alias("predicted_boolean_value"),
+    )
+
+    # Check the schema output:
+    validate_binary_classification_schema(predictions)
```

Comment:
I think this is great that you're doing this; but, I wonder if maybe it isn't actually helpful? Like, if this crashes, then you've just wasted the predictions work. Maybe MEDS-DEV should do this automatically after the predict step, that way the predictions are for sure generated, and we just check if they are valid after? Or, maybe we should modify meds-evaluation so this function can also make some changes and/or write it to disk similar to how MEDS-Transforms does in its `finalize_MEDS_*` stages? No change needed, just food for thought.

Reply:
@mmcdermott what you described happened in my local testing environment, where the test failed due to schema issues, it would be nice to have this separated.